src/acceptor/builder.rs

```diff
@@ -3,6 +3,9 @@ use std::sync::Arc;
 use hyper::server::conn::AddrIncoming;
 use rustls::ServerConfig;
 
+#[cfg(feature = "ring")]
```

Comment:
I don't think these `ring` guards make sense.

Reply:
```
warning: unused imports: `CertificateDer`, `PrivateKeyDer`
 --> src\acceptor\builder.rs:4:17
  |
4 | use pki_types::{CertificateDer, PrivateKeyDer};
  |                 ^^^^^^^^^^^^^^  ^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
```

This makes sense if ring is not part of the features

Reply:
Maybe I should just directly reference the type using full crate path.

Reply:
Maybe I should just directly reference the type using full crate path.

Reply:
IMO all of the `ring` guards in this module don't make sense. What errors do you get if you leave them out?

Reply:
> IMO all of the `ring` guards in this module don't make sense. What errors do you get if you leave them out?

A list of errors:
```
error[E0412]: cannot find type `CertificateDer` in this scope
  --> F:\.cache\cargo\registry\src\index.crates.io-6f17d22bba15001f\rustls-0.22.0-alpha.3\src\server\builder.rs:71:25
   |
71 |         cert_chain: Vec<CertificateDer<'static>>,
   |                         ^^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
1  + use pki_types::CertificateDer;
   |

error[E0412]: cannot find type `PrivateKeyDer` in this scope
  --> F:\.cache\cargo\registry\src\index.crates.io-6f17d22bba15001f\rustls-0.22.0-alpha.3\src\server\builder.rs:72:18
   |
72 |         key_der: PrivateKeyDer<'static>,
   |                  ^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this enum
   |
1  + use pki_types::PrivateKeyDer;
   |

error[E0412]: cannot find type `Error` in this scope
  --> F:\.cache\cargo\registry\src\index.crates.io-6f17d22bba15001f\rustls-0.22.0-alpha.3\src\server\builder.rs:73:31
   |
73 |     ) -> Result<ServerConfig, Error> {
   |                               ^^^^^ not found in this scope
   |
help: consider importing one of these items
   |
1  + use alloc::fmt::Error;
   |
1  + use core::error::Error;
   |
1  + use core::fmt::Error;
   |
1  + use crate::Error;
   |
     and 4 other candidates

error[E0412]: cannot find type `CertificateDer` in this scope
  --> F:\.cache\cargo\registry\src\index.crates.io-6f17d22bba15001f\rustls-0.22.0-alpha.3\src\server\builder.rs:89:25
   |
89 |         cert_chain: Vec<CertificateDer<'static>>,
   |                         ^^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
1  + use pki_types::CertificateDer;
   |

error[E0412]: cannot find type `PrivateKeyDer` in this scope
  --> F:\.cache\cargo\registry\src\index.crates.io-6f17d22bba15001f\rustls-0.22.0-alpha.3\src\server\builder.rs:90:18
   |
90 |         key_der: PrivateKeyDer<'static>,
   |                  ^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this enum
   |
1  + use pki_types::PrivateKeyDer;
   |

error[E0412]: cannot find type `Error` in this scope
  --> F:\.cache\cargo\registry\src\index.crates.io-6f17d22bba15001f\rustls-0.22.0-alpha.3\src\server\builder.rs:92:31
   |
92 |     ) -> Result<ServerConfig, Error> {
   |                               ^^^^^ not found in this scope
   |
help: consider importing one of these items
   |
1  + use alloc::fmt::Error;
   |
1  + use core::error::Error;
   |
1  + use core::fmt::Error;
   |
1  + use crate::Error;
   |
     and 4 other candidates
```

Reply:
Yep, guess that would do. But I think the bigger problem is that `rustls::ClientConfig` was not liberal enough as a builder...

Reply:
Maybe we should further remove ring as part of the default features...in exchange for a chaos crates that depends on hyper-rustls, but this would definitely promote a good use for custom crypto suites provider.

Reply:
No, we should mirror the API as it's setup in rustls alpha.4, which uses `builder()` with *ring* as a default and offers an alternative option to select a different provider.

Reply:
Acknowledged. Maybe this is the best we could do for now.